app/models/user.rb

```diff
@@ -24,6 +24,8 @@ class User < ApplicationRecord
   scope :confirmed, -> { where('confirmed_at IS NOT NULL') }
   scope :unconfirmed, -> { where('confirmed_at IS NULL') }
 
+  before_destroy :delete_mailchimp_user
```

Comment:
We have delete mailchimp user being done on a before_action, and update_mailchimp being called from the controller action. I don't really mind which one we do, but it would be good to be consistent

Reply:
`update_mailchimp` currently only handles subscribe and un-subscribing for the tags, so I'd need to add some logic in to be able to delete the user, hence going for a separate named function, but can merge them if you feel it'd be worth it. Also is more descriptive of what we do